File: src/game/Arma.js
export const Arma = Object.freeze({
  ESPADA: "ESPADA",
  MANDOBLE: "MANDOBLE",
  LANZA: "LANZA",
  ARCO: "ARCO",
  CATALIZADOR: "CATALIZADOR"
});

export function armaGanaA(arma, otra) {
  switch (arma) {
    case Arma.ESPADA: return otra === Arma.CATALIZADOR;
    case Arma.MANDOBLE: return otra === Arma.ESPADA;
    case Arma.LANZA: return otra === Arma.MANDOBLE;
    case Arma.ARCO: return otra === Arma.LANZA;
    case Arma.CATALIZADOR: return otra === Arma.ARCO;
    default: return false;
  }
}

export function armaRandom() {
  const values = Object.values(Arma);
  return values[Math.floor(Math.random() * values.length)];
}

File: java/src/game/Arma.java
package game;

public enum Arma {
    ESPADA, MANDOBLE, LANZA, ARCO, CATALIZADOR;

    public boolean ganaA(Arma otra) {
        return switch (this) {
            case ESPADA -> otra == CATALIZADOR;
            case MANDOBLE -> otra == ESPADA;
            case LANZA -> otra == MANDOBLE;
            case ARCO -> otra == LANZA;
            case CATALIZADOR -> otra == ARCO;
        };
    }

    public static Arma random() {
        Arma[] arma = values();
        return arma[(int) (Math.random() * arma.length)];
    }
}
